### PR TITLE
🏗 Add support for `gulp serve --coverage` coverage collection endpoint

### PR DESCRIPTION
**Pull request by rcebulko**
When `gulp serve` is called with the `--coverage` flag, a new `/coverage` endpoint will be available. Coverage data can be reported to that endpoint, and it provides a simple UI to browse coverage data throughout the codebase.

Sample of interface after loading and reporting from `examples/everything.amp.html`:
image

Note that currently the coverage data is still reported manually; doing so automatically, either within `gulp serve` or for `gulp e2e`, will be addressed separately. This PR just adds the flag and collection endpoint.

### CODE REVIEW

File: build-system/server/app.js
Comment (rsimha) [inline build-system/server/app.js:69]: Curious: Does this work with the various `gulp serve` flags like `--esm` or `--compiled` or `--sxg`?

**Review by rsimha — APPROVED** [review]

File: build-system/server/app.js
Comment (rcebulko) [reply build-system/server/app.js:69]: `--compiled` yes. As for `--esm` and `--sxg`, I haven't tried those yet, but I suspect they will be slightly more complicated based on some of the related code paths I saw. I'm going to work on `--esm` next week to make sure that's usable and the sourcemaps function properly. As for SxG, I don't have a deep enough understanding to know whether it would have any impact, and whether SxG would be any different from a coverage perspective

**Comment by rcebulko** [comment]
@ampproject/build-cop This failing test has been blocking this PR for almost two weeks now. Can it be skipped?
image

**Comment by zhouyx** [comment]
I have https://github.com/ampproject/amphtml/pull/30247 to fix it. Also
happy to skip the test if it's blocking.

**Comment by kristoferbaxter** [comment]
Let's skip it for now to unblock new PRs.

@rcebulko can you file a PR to skip until we figure it out?

**Comment by rcebulko** [comment]
> Let's skip it for now to unblock new PRs.
> 
> @rcebulko can you file a PR to skip until we figure it out?

Done

File: build-system/server/app.js
Comment (rsimha) [reply build-system/server/app.js:69]: Based on my understanding, this PR should have a similar effect on `--esm` and `--sxg`. So my guess is that if one of them works with the coverage middleware, both should. We can confirm once the initial code lands.

**Pull request merged**
